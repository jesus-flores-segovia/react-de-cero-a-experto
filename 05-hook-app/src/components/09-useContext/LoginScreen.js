import React, { useContext } from 'react';
import { useForm } from '../../hooks/useForm';
import { UserContext } from './UserContext';

export const LoginScreen = () => {

    const [values, handleInputChange, reset] = useForm({
        email: "",
        password: "",
    });

    const {email, password} = values;

    const {user, setUser} = useContext(UserContext);

    const handleSubmit = (e) => {
        e.preventDefault();
        setUser(values);
        reset();
        console.log(values);
    };

    return (
        <div>
            <h1>LoginScreen</h1>
            <hr/>
            <form onSubmit={handleSubmit}>
                <div className="mb-3">
                    <label htmlFor="inputEmail" className="form-label">Email address</label>
                    <input name="email" type="email" className="form-control" id="inputEmail" aria-describedby="emailHelp" value={email} onChange={handleInputChange}/>
                </div>
                <div className="mb-3">
                    <label htmlFor="inputPassword" className="form-label">Password</label>
                    <input name="password" type="password" className="form-control" id="inputPassword" value={password} onChange={handleInputChange}/>
                </div>
                <button type="submit" className="btn btn-primary">Login</button>
            </form>
        </div>
    )
}
