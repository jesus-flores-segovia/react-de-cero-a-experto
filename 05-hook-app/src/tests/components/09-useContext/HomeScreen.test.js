import { mount } from "enzyme";
import { HomeScreen } from "../../../components/09-useContext/HomeScreen";
import { UserContext } from "../../../components/09-useContext/UserContext";

describe("Tests inside file 'HomeScreen.test.js'", () => {
    
    const user = {
        name: "Jes√∫s Flores Segovia",
        email: "jesus@gmail.com"
    }

    const wrapper = mount(
    <UserContext.Provider value={{
        user
    }}>
        <HomeScreen/>
    </UserContext.Provider>
    )

    test("'HomeScreen' component must show correctly", () => {
        expect(wrapper).toMatchSnapshot();
    });
});