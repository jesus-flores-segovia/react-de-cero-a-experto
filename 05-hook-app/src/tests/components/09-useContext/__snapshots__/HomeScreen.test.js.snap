// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests inside file 'HomeScreen.test.js' 'HomeScreen' component must show correctly 1`] = `
<div>
  <h1>
    HomeScreen
  </h1>
  <hr />
  <pre>
    {
   "name": "Jes√∫s Flores Segovia",
   "email": "jesus@gmail.com"
}
  </pre>
</div>
`;
